ext/js/language/language-transforms.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+/**
+ * @param {string} inflectedSuffix
+ * @param {string} deinflectedSuffix
+ * @param {string[]} conditionsIn
+ * @param {string[]} conditionsOut
+ * @returns {import('language-transformer').SuffixRule}
+ */
+export function suffixInflection(inflectedSuffix, deinflectedSuffix, conditionsIn, conditionsOut) {
+    const suffixRegExp = new RegExp(inflectedSuffix + '$');
+    return {
+        type: 'suffix',
+        isInflected: suffixRegExp,
+        deinflected: deinflectedSuffix,
+        deinflect: (text) => text.slice(0, -inflectedSuffix.length) + deinflectedSuffix,
```

Comment:
If already using a Regex to find the inflected part, why not just use `String.prototype.replace()`?
Then it is possible to remove the `isInflected` field as well.

Reply:
(Just realized that this was originally a `replace`. What was the rationale behind the change 🤔

Reply:
It's because of some concerns about closures, see [here](https://github.com/themoeway/yomitan/pull/745#discussion_r1510004132) and [here](https://github.com/themoeway/yomitan/pull/745#discussion_r1510004331). I don't think there should be much of an impact on memory or performance either way, but at least the regex object can be avoided in this specific closure.